File: src/main/kotlin/com/nazarethlabs/notes/service/note/SearchNoteService.kt
package com.nazarethlabs.notes.service.note

import com.nazarethlabs.notes.dto.Note
import com.nazarethlabs.notes.helper.SearchHelper
import com.nazarethlabs.notes.repository.NoteStorageRepository

class SearchNoteService {
    fun filterNotes(filterText: String): List<Note> {
        val noteStorage = NoteStorageRepository()
        return SearchHelper.search(noteStorage.getAllNotes(), filterText)
    }
}

File: vscode/src/service/SearchNoteService.ts
import { Note } from '../dto/Note';
import { NoteRepository } from '../repository/NoteRepository';
import { FileHelper } from '../helper/FileHelper';
import { SearchHelper } from '../helper/SearchHelper';

export class SearchNoteService {
  private readonly repository: NoteRepository;

  constructor(repository: NoteRepository) {
    this.repository = repository;
  }

  search(query: string): Note[] {
    const allNotes = this.repository.getAllNotes();
    const contentMap = this.buildContentMap(allNotes);
    return SearchHelper.search(allNotes, query, contentMap);
  }

  private buildContentMap(notes: Note[]): Map<string, string> {
    const map = new Map<string, string>();
    for (const note of notes) {
      map.set(note.id, FileHelper.readText(note.filePath));
    }
    return map;
  }
}
